### ISSUE TITLE
[Shaman] Earth Shield

### ISSUE THREAD

**Issue by Marquito01** [dalaranwow/dalaran-wow#2667]
Healing from Earth Shield occurs right before the damage is taken (instead of right after), which results in almost constant overhealing from ES and renders this spell nearly useless. 

The previous topic was closed, but its still bugged: https://github.com/dalaranwow/dalaran-wow/issues/2263

Example:
Lets say tank has 40k HP, and will take 10k dmg. Earth shield will crit heal for 5k.

How it currently works: 40/40k HP -> 5k ES overheal -> 10k dmg = 30/40k HP
How it should work: 40/40k HP -> 10k dmg -> 5k ES heal = 35/40k HP

Basically, the spell needs a very small delay, so it heals after the hit is taken and not before.

**Comment by Zephyrino** [comment]
This is correct, and the same issue still occur's with a Priest's spell Prayer of Mending.

**Comment by Golganeth** [comment]
#3207
